### PR TITLE
Release 3.2.0

### PR DESCRIPTION
**Pull request by daven-gomes-cko**
## Proposed changes

Release 3.2.0 with a new `NetworkError` enum and new methods.

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [X] Bugfix (non-breaking change which fixes an issue)
* [X] New feature (non-breaking change which adds functionality)
* [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [X] Lint and unit tests pass locally with my changes
* [X] I have added tests that prove my fix is effective or that my feature works
* [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by daven-gomes-cko** [comment]
@deepesh-vasthimal-cko To be merged after https://github.com/checkout/frames-ios/pull/30

**Review by nicolas-maalouf-cko — APPROVED** [review]

**Comment by deepesh-vasthimal-cko** [comment]
@daven-gomes-cko Your other PR which is https://github.com/checkout/frames-ios/pull/30 has more updated changes do you want to merge that and then update this PR to get latest ?

**Review by deepesh-vasthimal-cko — CHANGES_REQUESTED** [review]
Your other PR which is #30 has more updated changes do you want to merge that and then update this PR to get latest ?

File: Source/CheckoutAPIClient.swift [hunk 1]
```diff
@@ -107,12 +108,48 @@ public class CheckoutAPIClient {
...
+        // swiftlint:disable:next force_try
+        var urlRequest = try! URLRequest(url: URL(string: url)!, method: HTTPMethod.post, headers: headers)
+        urlRequest.httpBody = try? jsonEncoder.encode(card)
+        request(urlRequest)
+            .validate().responseJSON { response in
+                let decoder = JSONDecoder()
+                switch response.result {
+                case .success:
+                    do {
+                        let cardTokenResponse = try decoder.decode(CkoCardTokenResponse.self, from: response.data!)
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutAPIClient.swift:131]: I remember you fixed all these force unwraps in other PR. Not sure why are they not reflecting here ? . Can you please check.

File: Source/CheckoutAPIClient.swift [hunk 2]
```diff
@@ -107,12 +108,48 @@ public class CheckoutAPIClient {
...
+                        completion(.success(cardTokenResponse))
+                    } catch let error {
+                        completion(.failure(.other(error: error)))
+                    }
+                case .failure(let responseError):
+                    do {
+                        let networkError = try decoder.decode(NetworkError.self, from: response.data!)
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutAPIClient.swift:138]: one more here : I remember you fixed all these force unwraps in other PR. Not sure why are they not reflecting here ? . Can you please check.

File: Source/CheckoutAPIClient.swift [hunk 3]
```diff
@@ -107,12 +108,48 @@ public class CheckoutAPIClient {
...
+    /// - parameter card: Card used to create the token
+    /// - parameter completion: Callback to execute if the request is successful or failed
+    public func createCardToken(
+        card: CkoCardTokenRequest,
+        completion: @escaping ((Swift.Result<CkoCardTokenResponse, NetworkError>) -> Void)
+    ) {
+        let url = "\(environment.urlPaymentApi)\(Endpoint.tokens.rawValue)"
+        let jsonEncoder = JSONEncoder()
+        // swiftlint:disable:next force_try
+        var urlRequest = try! URLRequest(url: URL(string: url)!, method: HTTPMethod.post, headers: headers)
```
Comment (deepesh-vasthimal-cko) [inline Source/CheckoutAPIClient.swift:123]: one more : I remember you fixed all these force unwraps in other PR. Not sure why are they not reflecting here ? . Can you please check.

**Pull request closed without merging**